homeassistant/components/device_tracker/unifi.py

```diff
@@ -0,0 +1,96 @@
+""" Supports scanning a unifi controller. """
+import logging
+from datetime import timedelta
+import threading
+
+from homeassistant.const import CONF_HOST, CONF_USERNAME, CONF_PASSWORD
+from homeassistant.util import Throttle
+from homeassistant.components.device_tracker import DOMAIN
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['unifi==1.2.3']
+
+CONF_PORT = "port"
+CONF_VERSION = "version"
+CONF_SITEID = "siteid"
+
+
+def get_scanner(hass, config):
+    """ Returns a unifi scanner. """
+
+    info = config[DOMAIN]
+
+    host = info.get(CONF_HOST)
+    username = info.get(CONF_USERNAME)
+    password = info.get(CONF_PASSWORD)
+    port = info.get(CONF_PORT)
+    version = info.get(CONF_VERSION)
+    siteid = info.get(CONF_SITEID)
+
+    scanner = UnifiDeviceScanner(host, username, password,
+                                 port, version, siteid)
+
+    return scanner if scanner.success_init else None
+
+
+class UnifiDeviceScanner(object):
+    """ This class queries a unifi wireless access controller. """
+    # pylint: disable=too-many-arguments
+    def __init__(self, host, username, password, port, version, siteid):
+        self.last_results = []
+
+        try:
```

Comment:
We can remove this construct, the `REQUIREMENTS` on top makes sure it is loaded. (It's a work in progress to remove this from other platforms too)